### PR TITLE
Moving activation method to an env variable

### PR DESCRIPTION
**Pull request by eyalkoren**

### CODE REVIEW

**Review by jackshirazi — APPROVED** [review]

File: charts/apm-attacher/values.yaml [hunk 1]
```diff
@@ -29,7 +29,8 @@ webhookConfig:
       image: docker.elastic.co/observability/apm-agent-java:latest
       artifact: "/usr/agent/elastic-apm-agent.jar"
       environment:
-        JAVA_TOOL_OPTIONS: "-javaagent:/elastic/apm/agent/elastic-apm-agent.jar -Delastic.apm.activation_method=K8S"
+        JAVA_TOOL_OPTIONS: "-javaagent:/elastic/apm/agent/elastic-apm-agent.jar"
+        ELASTIC_APM_ACTIVATION_METHOD: "K8S"
```
[the remark this answers is not in the document]
Comment (eyalkoren) [reply charts/apm-attacher/values.yaml]: Is this inconsistent with the other two on purpose?

[the remark this answers is not in the document]
Comment (eyalkoren) [reply charts/apm-attacher/values.yaml]: Anyway, your latest value seems to be translated to the correct one according to [spec](https://github.com/elastic/apm/blob/main/specs/agents/metadata.md#activation-method), I am just wondering why it was `K8S` before and whether the others are compliant with the current spec

[the remark this answers is not in the document]
Comment (jackshirazi) [reply charts/apm-attacher/values.yaml]: Previously I had enums for config and enums for reporting. The reporting ones were per spec, the config were deliberately different. Per your rec I've changed to one common enum under `common` so now the config value has to be the same as spec

[the remark this answers is not in the document]
Comment (eyalkoren) [reply charts/apm-attacher/values.yaml]: > now the config value has to be the same as spec

Not really, that's what I tried to say before- our config provides free mapping from string to enum and enums have properties, which can provide the exact decoupling you wanted between the enum/config name and the reported value (e.g. if the `ActivationType` enum has a `getIntakeValue()` method).

May other values be affected like this one?

[the remark this answers is not in the document]
Comment (jackshirazi) [reply charts/apm-attacher/values.yaml]: not sure I see the point in mapping the values now that we've moved to one enum and all these external implementors are still mutable. The others (aws lambda, fleet, remote attach, self attach) are all now set to use the spec values

**Pull request merged**